lib/src/front_end/ast_node_visitor.dart

```diff
@@ -1005,8 +1006,7 @@ class AstNodeVisitor extends ThrowingAstVisitor<Piece> with PieceFactory {
 
   @override
   Piece visitIndexExpression(IndexExpression node) {
-    Piece? targetPiece;
-    if (node.target case var target?) targetPiece = nodePiece(target);
+    var targetPiece = optionalNodePiece(node.target);
```

Comment:
Since I added `optionalNodePiece()` as a utility method in the PR, I went ahead and found every place that I could use it and swapped out the call.

Reply:
Could it be an extension method? `node.target?.nodePiece` reads clearly and the nullability is obvious that way.

Reply:
Oh, it's an instance method... nevermind.

Reply:
Yeah, there's some state in PieceWriter and CommentWriter that all of the front end AST -> Piece code needs to have implicit access to.